src/platform/datapath_winuser.c

```diff
@@ -4084,15 +4920,32 @@ CxPlatDataPathProcessCqe(
         break;
     }
     case CXPLAT_CQE_TYPE_SOCKET_IO: {
-        CXPLAT_SOCKET_PROC* SocketProc =
-            CONTAINING_RECORD(CxPlatCqeUserData(Cqe), CXPLAT_SOCKET_PROC, IoSqe);
-        if (Cqe->lpOverlapped == &SocketProc->IoSqe.Sqe) {
-            CxPlatDataPathSocketProcessReceiveCompletion(SocketProc, Cqe);
-        } else {
-            CxPlatDataPathSocketProcessSendCompletion(SocketProc, Cqe);
+        DATAPATH_IO_SQE* Sqe =
+            CONTAINING_RECORD(CxPlatCqeUserData(Cqe), DATAPATH_IO_SQE, DatapathSqe);
+
+        switch (Sqe->IoType) {
+        case DATAPATH_IO_RECV:
+        case DATAPATH_IO_CONNECTEX:
+        case DATAPATH_IO_ACCEPTEX:
+            CxPlatDataPathSocketProcessReceiveCompletion(Cqe, Sqe);
```

Comment:
I've started to tease apart the tangled IO completion model, but haven't gone so far as to refactor CxPlatDataPathSocketProcessReceiveCompletion and CxPlatDataPathSocketProcessSendCompletion into separate functions for separate IOs.